Sofa/Component/Engine/Generate/src/sofa/component/engine/generate/VolumeFromTetrahedrons.inl

```diff
@@ -0,0 +1,254 @@
+/******************************************************************************
+ *                 SOFA, Simulation Open-Framework Architecture                *
+ *                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+ *                                                                             *
+ * This program is free software; you can redistribute it and/or modify it     *
+ * under the terms of the GNU Lesser General Public License as published by    *
+ * the Free Software Foundation; either version 2.1 of the License, or (at     *
+ * your option) any later version.                                             *
+ *                                                                             *
+ * This program is distributed in the hope that it will be useful, but WITHOUT *
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+ * for more details.                                                           *
+ *                                                                             *
+ * You should have received a copy of the GNU Lesser General Public License    *
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+ *******************************************************************************
+ * Authors: The SOFA Team and external contributors (see Authors.txt)          *
+ *                                                                             *
+ * Contact information: contact@sofa-framework.org                             *
+ ******************************************************************************/
+#pragma once
+
+#include <sofa/config.h>
+#include <sofa/component/engine/generate/VolumeFromTetrahedrons.h>
+
+namespace sofa::component::engine::generate
+{
+
+using sofa::core::objectmodel::ComponentState;
+using sofa::helper::ReadAccessor;
+using sofa::type::vector;
+using sofa::core::ConstVecCoordId;
+using sofa::core::objectmodel::BaseData;
+
+
+template <class DataTypes>
+VolumeFromTetrahedrons<DataTypes>::VolumeFromTetrahedrons()
+    : d_positions(initData(&d_positions,"positions","If not set by user, find the context mechanical."))
```

Comment:
To be consistent with other components, the singular should be preferred (`position`).